internal/fwserver/attribute_plan_modification.go

```diff
@@ -49,11 +50,18 @@ func AttributeModifyPlan(ctx context.Context, a fwschema.Attribute, req tfsdk.Mo
 
 	var requiresReplace bool
 
+	privateProviderData := privatestate.EmptyProviderData(ctx)
+
+	if req.Private != nil {
+		privateProviderData = req.Private
+	}
```

Comment:
Does this need to be copied to `resp.Private` to prevent data loss? Unfortunately difficult to unit test currently since `tfsdk.Attribute` implements `fwxschema.AttributeWithPlanModifiers`, but we'll have other attribute implementations 🔜